src/Shader.cpp

```diff
@@ -8,7 +8,8 @@ std::vector<uint32_t>
 Shader::compile_sources(const std::vector<std::string>& sources,
                                    const std::vector<std::string>& files,
                                    const std::string& entryPoint,
-                                   std::vector<std::pair<std::string,std::string>> definitions) {
+                                   std::vector<std::pair<std::string,std::string>> definitions,
+                                   const TBuiltInResource resources) {
```

Comment:
Let's make this a const ref as the other params

Reply:
Ok